lib/msal-core/src/authority/B2cAuthority.ts

```diff
@@ -32,19 +32,11 @@ export class B2cAuthority extends AadAuthority {
     /**
      * Returns a promise with the TenantDiscoveryEndpoint
      */
-    public GetOpenIdConfigurationEndpointAsync(): Promise<string> {
-        const resultPromise = new Promise<string>((resolve, reject) =>
-            resolve(this.DefaultOpenIdConfigurationEndpoint));
-
-        if (!this.IsValidationEnabled) {
-            return resultPromise;
-        }
-
-        if (this.IsInTrustedHostList(this.CanonicalAuthorityUrlComponents.HostNameAndPort)) {
-            return resultPromise;
+    public async GetOpenIdConfigurationEndpointAsync(): Promise<string> {
+        if (!this.IsValidationEnabled || this.IsInTrustedHostList(this.CanonicalAuthorityUrlComponents.HostNameAndPort)) {
+            return this.DefaultOpenIdConfigurationEndpoint;
```

Comment:
this need to return a promise doesn't it ?

return new Promise.resolve(this.defaultEndpoint)

Reply:
Its an `async` function so it does inherently.

Reply:
Since it is an `async` function, all return statements are wrapped as promises from my understanding.

Reply:
ahh okay makes sense